dex/plan.sh

```diff
@@ -7,31 +7,39 @@ pkg_maintainer="Chef Software Inc. <support@chef.io>"
 pkg_license=("Apache-2.0")
 pkg_source="https://$gopkg"
 pkg_upstream_url=$pkg_source
-pkg_build_deps=()
 pkg_exports=(
   [port]=service.port
   [host]=service.host
 )
-pkg_deps=()
+pkg_deps=(core/glibc)
+pkg_build_deps=(core/go core/git core/gcc)
 pkg_bin_dirs=(bin)
-pkg_scaffolding=core/scaffolding-go
-scaffolding_go_base_path=github.com/dexidp
-scaffolding_go_build_deps=()
+
+do_before() {
+  GOPATH=$HAB_CACHE_SRC_PATH/$pkg_dirname
+}
 
 do_prepare() {
   build_line "GO_LDFLAGS=\"-w -X $gopkg/version.Version=v$pkg_version\""
   export GO_LDFLAGS="-w -X $gopkg/version.Version=v$pkg_version"
 }
 
 do_download() {
-  # `-d`: don't let go build it, we'll have to build this ourselves
-  build_line "go get -d $gopkg"
-  go get -d $gopkg
+  return 0
+}
+
+do_verify() {
+  return 0
+}
 
-  pushd "${scaffolding_go_gopath:?}/src/$gopkg"
+# Use unpack instead of download, so that plan-build can manage the
+# source path. This ensures us a clean checkout every time we build.
+do_unpack() {
+  git clone "$pkg_source" "$GOPATH/$gopkg"
+  ( cd "$GOPATH/$gopkg" || exit
```

Comment:
That's this new shellcheck rule, isn't it? 🤷‍♂️ Fine. I guess an appropriate "shellcheck disable" comment would have been fine, too?

Reply:
The rule has been around a while, but I think we could disable it in our CI.   Shellcheck will skip that check if run with `set -e`. hab-plan-build sets that, but shellcheck has no way to know that.  I'm going to open a PR on our CI code to change that, but for now I think I prefer the `||exit` instead of the Shellcheck disable.